### PR TITLE
Allow widget to operate as a searchbox for alternative launchers such as Aosp launcher3

### PR DESCRIPTION
**Pull request by paledega**
I added searchbox flag into AppWidgetProviderInfo

### CODE REVIEW

**Comment by paledega** [comment]
protected AppWidgetProviderInfo getSearchWidgetProvider() {
            SearchManager searchManager =
                    (SearchManager) getContext().getSystemService(Context.SEARCH_SERVICE);
            ComponentName searchComponent = searchManager.getGlobalSearchActivity();
            if (searchComponent == null) return null;
            String providerPkg = searchComponent.getPackageName();
            AppWidgetProviderInfo defaultWidgetForSearchPackage = null;
            AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(getContext());
            for (AppWidgetProviderInfo info : appWidgetManager.getInstalledProviders()) {
                if (info.provider.getPackageName().equals(providerPkg) && info.configure == null) {
                    if ((info.widgetCategory
                            & AppWidgetProviderInfo.WIDGET_CATEGORY_SEARCHBOX) != 0) {
                        return info;
                    } else if (defaultWidgetForSearchPackage == null) {
                        defaultWidgetForSearchPackage = info;
                    }
                }
            }
            return defaultWidgetForSearchPackage;
        }
    }

this is aosp launcher3 code part

**Comment by subsymbolic** [comment]
Thanks @paledega! Can you provide testing instructions/steps for this?

**Comment by paledega** [comment]
> Thanks @paledega! Can you provide testing instructions/steps for this?

Yes I can

File: app/src/main/res/xml/search_widget_info.xml [hunk 1]
```diff
@@ -26,4 +26,4 @@
     android:previewImage="@drawable/search_widget_preview"
     android:resizeMode="horizontal"
     android:updatePeriodMillis="86400000"
-    android:widgetCategory="home_screen|keyguard" />
+    android:widgetCategory="home_screen|keyguard|searchbox" />
```
Comment (subsymbolic) [inline app/src/main/res/xml/search_widget_info.xml]: Can you also make this change in `search_widget_info_light.xml`?

**Comment by subsymbolic** [comment]
Hi @paledega, thanks again for submitting this. I'm closing it as it hasn't progressed since my last comment. Feel free to resubmit a fresh PR when you're ready!

**Pull request closed without merging**